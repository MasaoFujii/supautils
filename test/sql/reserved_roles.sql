-- can't create a reserved role
create role reserved_but_not_yet_created;
-- can't create a reserved role, even if it exists
create role anon;
-- cannot rename an existing role to a reserved role
alter role fake rename to reserved_but_not_yet_created;
-- cannot rename a reserved role
alter role supabase_storage_admin rename to another;
-- cannot alter-options a reserved role
alter role supabase_storage_admin nologin superuser;
alter role supabase_storage_admin password 'pass';
-- cannot drop a reserved role
drop role fake, supabase_storage_admin;
drop role anon, fake;
