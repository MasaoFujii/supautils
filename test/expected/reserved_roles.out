-- cannot rename an existing role to a reserved role
alter role fake rename to reserved_but_not_yet_created;
ERROR:  The "reserved_but_not_yet_created" role is reserved, only superusers can rename it.
\echo

-- cannot rename a reserved role
alter role supabase_storage_admin rename to another;
ERROR:  The "supabase_storage_admin" role is reserved, only superusers can rename it.
\echo

-- cannot alter-options a reserved role
alter role supabase_storage_admin nologin superuser;
ERROR:  The "supabase_storage_admin" role is reserved, only superusers can alter it.
alter role supabase_storage_admin password 'pass';
ERROR:  The "supabase_storage_admin" role is reserved, only superusers can alter it.
\echo

-- cannot alter-config for a reserved role
alter role supabase_storage_admin set search_path to 'test';
ERROR:  The "supabase_storage_admin" role is reserved, only superusers can alter it.
\echo

-- cannot bypass alter-options check by using current_user
set role anon;
alter role current_user password 'pass';
ERROR:  The "anon" role is reserved, only superusers can alter it.
reset role;
\echo

-- cannot bypass alter-config check by using current_user
set role supabase_storage_admin;
alter role supabase_storage_admin set search_path to 'test';
ERROR:  The "supabase_storage_admin" role is reserved, only superusers can alter it.
reset role;
\echo

-- cannot drop a reserved role
drop role fake, supabase_storage_admin;
ERROR:  The "supabase_storage_admin" role is reserved, only superusers can drop it
drop role anon, fake;
ERROR:  The "anon" role is reserved, only superusers can drop it
\echo

-- cannot create a reserved role
create role reserved_but_not_yet_created;
ERROR:  The "reserved_but_not_yet_created" role is reserved, only superusers can create it.
\echo

-- cannot create a reserved role, even if it exists
create role anon;
ERROR:  The "anon" role is reserved, only superusers can create it.
